File: templates/template.java
/**
 * Problem: {PROBLEM_NAME}
 * Difficulty: {DIFFICULTY}
 * Date: {DATE}
 * Language: Java
 * Approach: {APPROACH}
 * Time: O({TIME_COMPLEXITY})
 * Space: O({SPACE_COMPLEXITY})
 */

public class {CLASS_NAME} {
    /**
     * {PROBLEM_DESCRIPTION}
     * 
     * @param {PARAM_TYPE} {PARAMETER} {PARAM_DESCRIPTION}
     * @return {RETURN_TYPE} {RETURN_DESCRIPTION}
     */
    public {RETURN_TYPE} {FUNCTION_NAME}({PARAM_TYPE} {PARAMETER}) {
        // TODO: Implement solution
        return null;
    }
    
    public static void main(String[] args) {
        {CLASS_NAME} solution = new {CLASS_NAME}();
        
        // Test cases
        Object[][] testCases = {
            // {{INPUT}, {EXPECTED_OUTPUT}},
        };
        
        for (int i = 0; i < testCases.length; i++) {
            Object input = testCases[i][0];
            Object expected = testCases[i][1];
            Object result = solution.{FUNCTION_NAME}(({PARAM_TYPE}) input);
            
            System.out.println("Test " + (i + 1) + ": " + 
                (result.equals(expected) ? "PASS" : "FAIL"));
            System.out.println("  Input: " + input);
            System.out.println("  Expected: " + expected);
            System.out.println("  Got: " + result);
        }
    }
}

File: templates/template.js
/**
 * Problem: {PROBLEM_NAME}
 * Difficulty: {DIFFICULTY}
 * Date: {DATE}
 * Language: JavaScript
 * Approach: {APPROACH}
 * Time: O({TIME_COMPLEXITY})
 * Space: O({SPACE_COMPLEXITY})
 */

/**
 * {PROBLEM_DESCRIPTION}
 * @param {{PARAM_TYPE}} {PARAMETER} - {PARAM_DESCRIPTION}
 * @return {{RETURN_TYPE}} {RETURN_DESCRIPTION}
 */
var {FUNCTION_NAME} = function({PARAMETERS}) {
    // TODO: Implement solution
};

// Test cases
const testCases = [
    // [{INPUT}, {EXPECTED_OUTPUT}],
];

testCases.forEach((testCase, index) => {
    const [input, expected] = testCase;
    const result = {FUNCTION_NAME}(input);
    console.log(`Test ${index + 1}: ${JSON.stringify(result) === JSON.stringify(expected) ? 'PASS' : 'FAIL'}`);
    console.log(`  Input: ${JSON.stringify(input)}`);
    console.log(`  Expected: ${JSON.stringify(expected)}`);
    console.log(`  Got: ${JSON.stringify(result)}`);
});
